2nd-gen/thumbnails/functions/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+// [START v2storageImports]
+// [START v2storageSDKImport]
+const { onFinalize} = require('firebase-functions/v2/storage');
+// [END v2storageSDKImport]
+
+// [START v2storageAdditionalImports]
+const { initializeApp, storage} = require ('firebase-admin/app');
```

Comment:
Looks like you're trying to use the modular admin SDK for Cloud Storage.
I believe it should be `getStorage` instead of `storage` (as shown in the [docs](https://firebase.google.com/docs/admin/migrate-node-v10#using-v10-modular-entry-points)):

```suggestion
const {onFinalize} = require('firebase-functions/v2/storage');
// [END v2storageSDKImport]

// [START v2storageAdditionalImports]
const {initializeApp} = require ('firebase-admin/app');
const {getStorage} = require('firebase-admin/storage');
```

Reply:
+1, and note that you'll need to remove the space in the `initializeApp` require:

```diff
- const {initializeApp} = require ('firebase-admin/app');`
+ const {initializeApp} = require('firebase-admin/app');`
```